cmd/gpu-kubelet-plugin/driver.go

```diff
@@ -177,6 +202,68 @@ func (d *driver) nodeUnprepareResource(ctx context.Context, claimNs kubeletplugi
 	return nil
 }
 
+func (d *driver) deviceHealthEvents(ctx context.Context, nodeName string) {
+	klog.Info("Processing device health notifications")
+	for {
+		select {
+		case <-ctx.Done():
+			klog.V(6).Info("Stop processing device health notifications")
+			return
+		case device, ok := <-d.deviceHealthMonitor.Unhealthy():
+			if !ok {
+				klog.V(6).Info("Health monitor channel closed")
+				return
+			}
+			uuid := device.UUID()
+
+			klog.Warningf("Received unhealthy notification for device: %s", uuid)
+
+			if !device.IsHealthy() {
+				klog.V(6).Infof("Device: %s is aleady marked unhealthy. Skip republishing resourceslice", uuid)
```

Comment:
In practice, how often could we see a log message like this?

What I see here right now: we can get the `d.deviceHealthMonitor.Unhealthy()` event multiple times, even if we had already processed that device before. I wonder how often we should expect that to happen.

Reply:
there can be events burst. @lalitadithya showed me logs of device-plugin xid errors in a cluster which clearly showed same event logged multiple times. 

@lalitadithya is it possible to share the log here.